File: examples/pingpong_demo/ping/ping.go
package main

import (
	"encoding/json"
	"fmt"
	"io"
	"log"

	"context"

	dappclient "inspr.dev/inspr/pkg/client"
	"inspr.dev/inspr/pkg/sidecars/models"
)

func main() {

	client := dappclient.NewAppClient()
	ctx, cancel := context.WithCancel(context.Background())
	defer cancel()

	sentMsg := "Ping!"
	if err := client.WriteMessage(ctx, "pingoutput", sentMsg); err != nil {
		fmt.Printf("an error occurred: %v", err)
		return
	}
	client.HandleChannel(
		"pinginput",
		func(ctx context.Context, body io.Reader) error {
			var ret models.BrokerMessage

			decoder := json.NewDecoder(body)
			if err := decoder.Decode(&ret); err != nil {
				return err
			}

			fmt.Println(ret)

			if err := client.WriteMessage(ctx, "pingoutput", sentMsg); err != nil {
				fmt.Println(err)
				return err
			}
			return nil
		},
	)
	log.Fatal(client.Run(ctx))
}

File: clients/python/example/pingpong/ping/ping.py
from inspr import *
import sys

PING_INPUT_CHANNEL = "pinginput"
PING_OUTPUT_CHANNEL = "pingoutput"

def main():
    client = Client()
    msg = "Ping!"

    @client.handleChannel(PING_INPUT_CHANNEL)
    def readPongAndSendPing(data):

        if data == 'Pong!':
            print(data, file=sys.stderr)
        else:
            print('Not received Pong', file=sys.stderr)

        try:
            client.writeMessage(PING_OUTPUT_CHANNEL, msg)
        except:
            raise Exception


    try:
        client.writeMessage(PING_OUTPUT_CHANNEL, msg)
    except:
        print("An error has occured", file=sys.stderr)
        return

    client.run()

if __name__ == "__main__":
    main()
